pkg/input/remoteread/remoteread.go

```diff
@@ -1,3 +1,242 @@
 package remoteread
 
-// TBD, help wanted!
+import (
+	"context"
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/storage/remote"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+	"github.com/thanos-community/obslytics/pkg/input"
+	"github.com/thanos-community/obslytics/pkg/version"
+	"net/url"
+
+	"time"
+)
+
+type remoteReadInput struct {
+	logger log.Logger
+	conf   input.InputConfig
+}
+
+func NewRemoteReadInput(logger log.Logger, conf input.InputConfig) (remoteReadInput, error) {
+	return remoteReadInput{logger: logger, conf: conf}, nil
+}
+
+// TranslatePromMatchers returns proto matchers (prompb) from Prometheus matchers.
+// NOTE: It allocates memory.
+func TranslatePromMatchers(ms ...*labels.Matcher) ([]*prompb.LabelMatcher, error) {
```

Comment:
What about using "promclient.Translate..." function from Thanos repo?

Reply:
The Thanos function converts it into a `storepb.LabelMatcher` we need the object in `prompb.LabelMatcher` type.

Reply:
Cool, we can play with that later.